iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -126,21 +134,154 @@ ProcessManager::ProcessManager(RouDiMemoryInterface& roudiMemoryInterface,
     m_memoryManagerOfCurrentProcess = m_segmentInfo.m_memoryManager;
 }
 
-void ProcessManager::killAllProcesses() noexcept
+void ProcessManager::killAllProcesses(const units::Duration finalKillTime) noexcept
 {
     std::lock_guard<std::mutex> g(m_mutex);
+    cxx::vector<bool, MAX_PROCESS_NUMBER> processStillRunning(m_processList.size(), true);
+    int i = 0;
+
+    auto yield = []() { std::this_thread::sleep_for(std::chrono::milliseconds(250)); };
 
     // send SIGTERM to all running applications
-    typename ProcessList_t::iterator it = m_processList.begin();
-    const typename ProcessList_t::iterator itEnd = m_processList.end();
+    for (auto& process : m_processList)
+    {
+        // if it was killed we need to check if it really has terminated, if we can't kill it, we consider it
+        // terminated
+        processStillRunning[i] = requestShutdownOfProcess(process, ShutdownPolicy::SIG_TERM, ShudownLog::FULL);
+        ++i;
+    }
 
-    for (; itEnd != it; ++it)
+    // wait for process to complete
+    bool allProcessFinished = false;
+    posix::Timer finalKillTimer(finalKillTime);
```

Comment:
Can you please avoid using the posix timer since we want to scrap it ?

Reply:
the posix timer is schizophrenic and has a good and a bad part. This one uses the good part, which should be refactored out as `CountdownTimer` or `DeadlineTimer`

Reply:
I think the name `StopWatch` fits best ;-)

Reply:
Ok, so what do you guys think? Keep it for now and refactor later when a countdown timer is implemented or use some sleeps and chrono right now?

Reply:
@mossmaurice as stop watch counts up, that timer counts down ;)

Reply:
I agree with @elBoberido 's definition :) 
Timer refactoring is #337.